src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -49,7 +49,7 @@
 
   public boolean isConnected()
   {
-    return key.isValid() && ((SocketChannel)key.channel()).isConnected();
+    return key != null && key.isValid() && ((SocketChannel)key.channel()).isConnected();
```

Comment:
Vlad - do you want to send another PR for just this check so that the build passes consistently?